### PR TITLE
Rebate Flows

### PR DESCRIPTION
**Pull request by nitaliano**
**Description**

* Adds a `HexBadge` component used for displaying any sort of hex with a label and the shortened version of the hex
* Adds a `NetworkBadge` component that takes in a chain id and will display the chain name and it's logo
* Adds a `Network` component this is a general purpose component used that displays the chain name and logo in a flex row, it's being used in other components that need network info
* Adds a `superchain.ts` file that contains all the superchain defs. We'll eventually move this to a common pkg.
* Updates contract creation flow to use relations returned in request
* Updates rebate flows to be integrated with backend except the claim mutation
* Update our api types to be inferred from the backend, still not crazy about how it's being inferred something we'll need to address so we can pull these types from the backend instead of inference for our primitive objects.

### CODE REVIEW

File: apps/dapp-console/app/settings/contracts/page.tsx [hunk 1]
```diff
@@ -6,8 +6,17 @@ import { ClaimedRebateProgressBanner } from '@/app/settings/components/ClaimedRe
 import { CoinbaseVerificationBanner } from '@/app/settings/components/CoinbaseVerificationBanner'
 import { DeployedApps } from '@/app/settings/contracts/DeployedApps'
 import { ClaimedRebates } from '@/app/settings/components/ClaimedRebates'
+import { apiClient } from '@/app/helpers/apiClient'
+import { useEffect } from 'react'
 
 export default function Contracts() {
+  const { mutate: syncCbVerification } =
```
Comment (nitaliano) [inline apps/dapp-console/app/settings/contracts/page.tsx:13]: Adding this sync so it always happens on the contract page, so people can get unstuck with a reload. I think we should look into sharing the cb verified code so the FE can start polling this when we have the rebate dialog modal open. We can poll EAS from the FE and when we find an attestation we call this sync from the FE.

File: apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx [hunk 2]
```diff
@@ -8,39 +8,50 @@ import {
...
-  contractAddress,
-  contractDeploymentTransaction,
-  amountClaimed = BigInt(0),
-  isVerified = false,
+  contract,
   children,
 }: ClaimRebateDialogProps) => {
   const [isOpen, setOpen] = useState(false)
 
+  const { data: wallets } = apiClient.wallets.listWallets.useQuery({})
```
Comment (tremarkley) [inline apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx:32]: if you're just looking for wallet verifications, the `walletVerifications` endpoint might be better here: https://github.com/ethereum-optimism/ecosystem/blob/dfaba331691e610cdcbb4154b43778007709e17e/apps/dapp-console-api/src/routes/wallets/WalletsRoute.ts#L128

File: apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx [hunk 3]
```diff
@@ -8,39 +8,50 @@ import {
...
+  const { data: totalRebateAmount } =
+    apiClient.Rebates.totalRebatesClaimed.useQuery()
+  const amountClaimed = useMemo(
+    () => totalRebateAmount ?? BigInt(0),
+    [totalRebateAmount],
+  )
+
   const gasFeeToReimburst = useMemo(() => {
-    const gasAmount = contractDeploymentTransaction.gas
+    const gasAmount = BigInt(contract.transaction?.gasUsed ?? 0)
```
Comment (tremarkley) [inline apps/dapp-console/app/settings/components/ClaimRebateDialog.tsx:46]: should this be gasUsed * gasPrice? that is how we calculate it on the backend: https://github.com/ethereum-optimism/ecosystem/blob/dfaba331691e610cdcbb4154b43778007709e17e/apps/dapp-console-api/src/routes/rebates/RebatesRoute.ts#L261

**Review by tremarkley — APPROVED** [review]

File: apps/dapp-console/app/components/Selects/L2NetworkSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  onNetworkChange,
+}: SuperchainDropdownProps) => {
+  const [selectedNetwork, setSelectedNetwork] = useState<L2NetworkSelectItem>({
+    id: optimism.id,
+    chain: optimism,
+    config: superchain.op,
+  })
+
+  const allChains = useMemo(() => {
+    return Object.values(superchain).reduce((arr, config) => {
```
Comment (tremarkley) [inline apps/dapp-console/app/components/Selects/L2NetworkSelect.tsx:36]: should this be filtered to only include l2s? I think with the way it is written now l1 eth would be included in this.

**Pull request merged**